test/apps/iodemo/io_demo.cc

```diff
@@ -1196,10 +1193,21 @@ class DemoClient : public P2pDemoCommon {
     }
 
     server_info_t &get_server_info(const UcxConnection *conn) {
-        const size_t server_index = get_server_index(conn);
+        const size_t server_index = get_active_server_index(conn);
 
-        assert(server_index < _server_info.size());
-        return _server_info[server_index];
+        if (server_index < _server_info.size()) {
+            return _server_info[server_index];
+        }
+
+        for (size_t i = 0; i < _server_info.size(); ++i) {
+            if (_server_info[i].conn == conn) {
+                return _server_info[i];
+            }
+        }
```

Comment:
we should avoid calling get_server_info if a connection was not established. where is it called?
instead, can we save server_index in DisconnectCallback?